hugofs/glob.go

```diff
@@ -55,7 +55,7 @@ func Glob(fs afero.Fs, pattern string, handle func(fi FileMetaInfo) (bool, error
 			return nil
 		}
 
-		if g.Match(p) {
+		if g.Match(strings.ToLower(p)) {
```

Comment:
I have a few comments:

* There is a filenamesGlobCache in the glob package that should be used for this (it's used in other similar situations, not sure why not here).
* I do notice, however, that I the `isCaseSensitive` is always set to `false`, and I don't remember why... Changing that value without doing additional checks sounds dangerous.

**Using upper case in filename is something I would highly discourage, but I suspect that comes from the Windows world.**

What I suggest is this, and please arrest me if I'm thinking wrong:

* In globCache.GetGlob we check if the pattern contains any upper case chars.
* If so, we return a `globDecorator` with 2 globs (one all lower case)
* The match Match would then do one or two checks and return the first positive match.
* With that we can remove the isCaseSensitive flag which isn't in use anyway.


Also, if the /commands package tests keep failing on GitHub, do a if htesting.IsCi skip so we can get the tests here green.

Reply:
Re. failing test, see https://github.com/gohugoio/hugo/commit/08f0984f918bb97d35c1765d3b94b1a9a1a3b9cd

Reply:
> Using upper case in filename is something I would highly discourage, but I suspect that comes from the Windows world.

I want to say I am in the same boat ... but, you know, we are sometimes in the undesirable situation against our will.

Thank you for your suggestion. I would follow it mostly.